libs/filabridge/include/private/filament/UibStructs.h

```diff
@@ -227,6 +226,14 @@ struct PerRenderableUibBone { // NOLINT(cppcoreguidelines-pro-type-member-init)
 static_assert(CONFIG_MAX_BONE_COUNT * sizeof(PerRenderableUibBone) <= 16384,
         "PerRenderableUibBone exceed max UBO size");
 
+struct alignas(16) PerRenderableMorphingUib {
+  static constexpr utils::StaticString _name{ "MorphingUniforms" };
+  int count;
+  math::float4 weights[CONFIG_MAX_MORPH_TARGET_COUNT];
+};
```

Comment:
Indentation is 4 spaces. Definitely don't use `int` here because that's 8 bytes. Use `uint32_t`. Make sure to make the alignment apparent. Either use alignas(16) or  use padding.

I'm confused here, why do we have 4 floats per morph target?

Also I think we should use 16 bits float for the weights. So I think it should be something like:

`math::uint4 weights[CONFIG_MAX_MORPH_TARGET_COUNT / 8];  // 8 weights per "line"`

Or am I missing something?

Reply:
Because the uniform block follows std140 memory layout. In std140, the array stride(the bytes between array elements) is always rounded up to the size of a vec4.

Reply:
Right, right. But the way to fix this is to unroll the loop 4 (or 8 times) in the shader. Something like:

```
for (i=0 ; i<count ; i+=4) {
  float w[4] = { weight[i/4].x, weight[i/4].y, weight[i/4].z, weight[i/4].w };
  for (j=0 ; j<4 && i+j < count ; j++) {
     // use w[j] here
  }
}
```

I can live with this for now, but please have a comment here (and in the shader) that explains the situation.

Reply:
Add a comment and your ways looks very good. I never thought your way before.